### PR TITLE
Make InsertObserverPass work on module level.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19386 Make InsertObserverPass work on module level.**
* #19385 Clang-format torch/csrc/jit/passes/quantization.cpp.

Differential Revision: [D14991916](https://our.internmc.facebook.com/intern/diff/D14991916)

### CODE REVIEW

**Review by nishantpdce — CHANGES_REQUESTED** [review]

File: torch/csrc/jit/init.cpp
Comment (ZolotukhinM) [reply torch/csrc/jit/init.cpp]: > why not use graph contained in top level module method? Does it not make difference

Because we don't need this node to exist in any actual graph - we just use it to copy from and then delete. Creating a temporary graph for this seems 1) easier, 2) better aligned with the semantics.

File: torch/csrc/jit/passes/quantization.cpp
Comment (ZolotukhinM) [reply torch/csrc/jit/passes/quantization.cpp]: Graph has access to inputs and outputs as well. This pass is supposed to be as simple as possible, I don't think we're going to add specialized handling of arguments into it.

Comment (nishantpdce) [reply torch/csrc/jit/passes/quantization.cpp]: we will need in following change. graph inputs() does not give information about method external inputs which is required to add observers of type activation. So good to pass method here as argument

Comment (ZolotukhinM) [reply torch/csrc/jit/passes/quantization.cpp]: This pass doesn't work with graph inputs, it only adds observers for values produced by internal nodes of the graph. Could you please elaborate on what exactly you mean?

File: torch/csrc/jit/passes/quantization.h
Comment (zdevito) [inline torch/csrc/jit/passes/quantization.h]: This API looks like it copies the module, but it is mutating the module. If the intention is to mutate it would be better for this to be `void`.

**Review by zdevito — APPROVED** [review]
Seems fine. Some minor comments.

File: torch/csrc/jit/passes/quantization.h
Comment (ZolotukhinM) [reply torch/csrc/jit/passes/quantization.h]: The idea is that it should copy the module, it's just not implemented yet.

I currently have second thoughts about landing this - maybe I'll wait until all module/compilation_unit diffs land and then fix TODO in this one. It might also make sense to allow to specify which method to run on on the API if we don't want observers everywhere.

File: test/test_jit.py
Comment (eellison) [inline test/test_jit.py]: Is this just for testing? This pass relies on Values's unique name, which is not guaranteed to be stable @ZolotukhinM @nishantpdce

Comment (ZolotukhinM) [reply test/test_jit.py]: @eellison, this pass doesn't rely on the names. It records the names, but only for better debugability. It would work if we use any unique id, and it doesn't have to be the name (but using name allows us to more conveniently examine the data when the names are available).

If we "break" the names at any point, we should just remove the corresponding check from the test - please let me know if you think we need to do it now. The pass itself would not break from that.

Comment (eellison) [reply test/test_jit.py]: Yes, we should remove all calls to uniqueName() as something that affects output. I made unrelated changes and the quantization tests broke as a result

Comment (ZolotukhinM) [reply test/test_jit.py]: No problem, I'll remove it from the test.

Comment (ZolotukhinM) [reply test/test_jit.py]: Here is the PR removing these checks: #21657

**Comment by ZolotukhinM** [comment]
This is no longer relevant.

**Pull request closed without merging**
